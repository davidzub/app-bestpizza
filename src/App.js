import React from "react";
import { Container, Col, Row } from "react-bootstrap";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import PrivateRoute from "./Components/PrivateRoute";
import LoginComponent from "./Components/Pages/LoginComponent";
import ShopsComponent from "./Components/Pages/ShopsComponent";
import LeftComponent from "./Components/Left/LeftComponent";
import ProductsComponent from "./Components/Pages/ProductsComponent";
import AlertComponent from "./Components/General/AlertComponent";
import "./styles/App.css";
function App() {
  return (
    <Router>
      <main>
        <Container fluid className="background">
          <Row>
            <Col xs={12} md={5}>
              <LeftComponent />
            </Col>
            <Col xs={12} md={7} className="wrapper">
              <Switch>
                <PrivateRoute path="/shops" component={ShopsComponent} />
                <PrivateRoute
                  path="/products/:id"
                  component={ProductsComponent}
                />
                <Route path="/" component={LoginComponent} />
              </Switch>
            </Col>
          </Row>
          <AlertComponent />
        </Container>
      </main>
    </Router>
  );
}

export default App;
