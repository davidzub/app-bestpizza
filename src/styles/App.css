*{
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    font-family: 'Roboto', sans-serif;
    color: #707070;
}
.background{
    background-image: url('./../images/imagebkg.png');
    background-size: cover;
    background-attachment: fixed;
    min-height: 100vh;
  }
  .wrapper{
    background-color: #fff;
    min-height: 100vh;
  }
  @media (max-width: 1024px) {
    .wrapper{
      margin-bottom: 60px;
    }
  }