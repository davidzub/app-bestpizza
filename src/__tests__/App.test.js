import React from "react";
import { shallow } from "enzyme";
import App from "../App";

describe("<App />", () => {
  let wrapper
  beforeAll(() => {
    wrapper = shallow(
        <App />
    );
  });
  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Validate element 'wrapper'", () => {
    expect(wrapper.find(".wrapper")).toHaveLength(1);
  });
});
