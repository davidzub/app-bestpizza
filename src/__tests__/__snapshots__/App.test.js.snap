// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render 1`] = `
<BrowserRouter>
  <main>
    <Container
      className="background"
      fluid={true}
    >
      <Row
        noGutters={false}
      >
        <Col
          md={5}
          xs={12}
        >
          <LeftComponent />
        </Col>
        <Col
          className="wrapper"
          md={7}
          xs={12}
        >
          <Switch>
            <PrivateRoute
              component={[Function]}
              path="/shops"
            />
            <PrivateRoute
              component={[Function]}
              path="/products/:id"
            />
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>
        </Col>
      </Row>
      <AlertComponent />
    </Container>
  </main>
</BrowserRouter>
`;
